### PR TITLE
Ignore / Force_index

### PR DESCRIPTION
**Pull request by adelavega**
- Renames arguments to be pybids compliant
- If string starts and ends with `/` then it is compiled using `re`

### CODE REVIEW

File: fitlins/cli/run.py [hunk 1]
```diff
@@ -88,10 +88,10 @@ def get_parser():
                         choices=['MNI152NLin2009cAsym', ''],
                         default='MNI152NLin2009cAsym',
                         help='registered space of input datasets. Empty value for no explicit space.')
-    g_bids.add_argument('--include', action='store', default=None,
+    g_bids.add_argument('--force_index', action='store', default=None,
```
Comment (effigies) [inline fitlins/cli/run.py]:
```suggestion
    g_bids.add_argument('--force-index', action='store', default=None,
```

File: fitlins/interfaces/bids.py [hunk 2]
```diff
@@ -161,8 +161,20 @@ class LoadBIDSModel(SimpleInterface):
...
-        ignore = self.inputs.ignore or None
+        if self.inputs.ignore:
+            ignore = []
+            for patt in self.inputs.ignore:
+                # If starts and ends with `/` `/`, treat as aregex
+                if patt[0] == '/' and patt[-1] == '/':
+                    patt = re.compile(patt[1:-1])
+                ignore.append(patt)
+        else:
+            ignore = None
```
Comment (effigies) [inline fitlins/interfaces/bids.py]: Just to be a little more concise, the whole `if/else` block can be a list comprehension:

```Python
ignore = [
    # If entry looks like `/<pattern>/`, treat `<pattern>` as a regex
    re.compile(ign[1:-1]) if (ign[0], ign[-1]) == ('/', '/') else ign
    # Iterate over empty tuple if undefined
    for ign in self.inputs.ignore or ()]
```

File: fitlins/interfaces/bids.py [hunk 3]
```diff
@@ -161,8 +161,20 @@ class LoadBIDSModel(SimpleInterface):
...
+
+        # If empty , then None
```
Comment (effigies) [inline fitlins/interfaces/bids.py]:
```suggestion
        # If empty, then None
```

File: fitlins/interfaces/bids.py (hunk 2 above)
Comment (adelavega) [reply fitlins/interfaces/bids.py]: I considered that but didn't want to be too Rubyesque. But the comments help a lot.

Comment (effigies) [reply fitlins/interfaces/bids.py]: Yeah, each line doesn't take much work to understand, but the two together are pretty dense without the comments. But I loathe `for: append()` loops.

Comment (adelavega) [reply fitlins/interfaces/bids.py]: Same. I think yours is better though so I'll add it.

File: fitlins/cli/run.py [hunk 4]
```diff
@@ -88,10 +88,10 @@ def get_parser():
                         choices=['MNI152NLin2009cAsym', ''],
                         default='MNI152NLin2009cAsym',
                         help='registered space of input datasets. Empty value for no explicit space.')
-    g_bids.add_argument('--include', action='store', default=None,
+    g_bids.add_argument('--force-index', action='store', default=None,
                         help='regex pattern to include files')
```
Comment (effigies) [inline fitlins/cli/run.py]: These are no longer regexes, right? Just paths?

Comment (adelavega) [reply fitlins/cli/run.py]: Actually, they're both regexes so we should apply the same logic

Comment (adelavega) [reply fitlins/cli/run.py]: You are a strong contender for code reviewer of the year 🏅

Comment (adelavega) [reply fitlins/cli/run.py]: You get an emoji medal. Treasure it.

Comment (effigies) [inline fitlins/cli/run.py]:
```suggestion
                        help='regex pattern or string to include files')
```

**Pull request merged**